Semigroup-Monoid.md

```diff
@@ -0,0 +1,318 @@
+
+---
+
+# Semigroup :: combine
+
+`Semigroup` abstracts the ability to __combine__ two values of the same type `A`.
+
+In other words, it provides a combining function for the `A` type:
+
+```kotlin
+fun A.combine(b: A): A
+```
+
+^ The most important behaviour provided by Semigroup is the combine function.
+^ With combine, we can combine/concat/join two A values.
+^ Semigroup defines on a type A a combination function which takes as input two elements of A to return an element of the same type A.
+
+---
+
+# Semigroup :: Data types
+
+Any type constructor whose contents can be combined can provide an instance of `Semigroup`.
+
+- `Option<A>`
+- `NonEmptyList<A>`
+- `ListK<A>`
+- `SequenceK<A>`
+- `SetK<A>`
+- `MapK<A>`
+- `SortedMapK<A>`
+- `Either<A>`
+- `Try<A>`
+
+Arrow also offers Semigroup instances to native Kotlin `String`, `Byte`, `Double`, `Int`, `Long`, `Short` and `Float` types.
+
+^ In order for a data type content to be “combinable”, we need its type constructor to be able to provide an instance of a Semigroup. Some data types examples on Arrow that offer exactly that would be: Option, NonEmptyList, ListK, SequenceK, SetK...
+
+---
+
+# Semigroup :: combine
+
+In this example we are showing the main characteristic about `Semigroup`, the `combine` method.
+As you can see here, we can apply it for both a `Double` and a `String` type, but remember you can use it with many other types.
+
+
+```kotlin
+with(Double.semigroup()){
+    4.5.combine(7.0).combine(2.5)
+}
+//14.0
+```
+
+```kotlin
+with(String.semigroup()){
+    "This is ".combine("functional ") + ("programming.")
+}
+//This is functional programming.
+```
+
+Additionaly `Semigroup` adds `+` syntax to all types for which a `Semigroup` instance exists
+
+^ This is an example which explain the main characteristic about Semigroup´s combine method in a clear way.
+^ We are applying it to a Double and a String in this case, but remember you can use it with many other types.
+^ Additionaly Semigroup adds + syntax (mark +) to all types for which a Semigroup instance exists
+^ With a semigroup, we gain the capability to combine, and that is the main value of it.
+
+---
+
+# Semigroup :: Laws
+
+All Typeclasses __must satisfy some mathematical laws__ in order to be considered a Typeclass.
+
+Arrow has tests in place for those laws so typeclasses can keep their integrity over time.
+
+^ When we think about typeclasses, we assume they need to satisfy some mathematical laws in order to be considered a typeclass.
+^ Those laws are coded and enforced by tests inside Arrow, so we can ensure typeclasses integrity over time.
+^ Some laws could be identity, associativity, composition, and so on.
+
+---
+
+# Semigroup :: Laws
+
+Semigroup satisfies the following law:
+
+- __Associativity__: When adding or multiplying, it doesn't matter how we combine the group of numbers.
```

Comment:
This property is universal and not just for numbers

Reply:
Sure!